.github/workflows/workflow.yml

```diff
@@ -1,36 +1,53 @@
-# For most projects, this workflow file will not need changing; you simply need
-# to commit it to your repository.
-#
-# You may wish to alter this file to override the set of languages analyzed,
-# or to provide custom queries or build logic.
-#
-# ******** NOTE ********
-# We have attempted to detect the languages in your repository. Please check
-# the `language` matrix defined below to confirm you have the correct set of
-# supported CodeQL languages.
-#
-name: 'CodeQL'
+name: Workflow
 
 on:
-  push:
-    branches: ['develop', main]
   pull_request:
-    # The branches below must be a subset of the branches above
-    branches: ['develop']
+    branches:
+      - '**'
   schedule:
     - cron: '35 9 * * 6'
 
 jobs:
+  build:
+    runs-on: ubuntu-latest
+    if: github.event_name != 'schedule'
+    strategy:
+      matrix:
+        node-version: [22.x]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: npm install
+      - run: npm run check
+  test:
+    runs-on: ubuntu-latest
+    if: github.event_name != 'schedule'
+    needs: build
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install system dependencies
+        run: |
+          echo 0 | sudo tee /proc/sys/kernel/apparmor_restrict_unprivileged_userns
+      - name: Install project dependencies
+        run: yarn install
+
+      - name: Run tests
+        run: yarn test
```

Comment:
Also, why is test and build in a separate section, cant both be done in a single section

Reply:
As discussed earlier, this is to keep them as separate jobs to ensure the separation of concerns.

Reply:
Please run the tests after the build steps, I don't see any point in running them separately

Reply:
Acknowledged.

Reply:
I have moved this to one section.